dev-team/hooks/validate-gate-progression.sh

```
@@ -299,21 +299,63 @@ fi
+      *"Gate 6"*)    recovery_steps+=("Gate 6: Load Skill(ring:dev-integration-testing), dispatch ring:qa-analyst (test_mode=integration, write only)") ;;
+      *"Gate 7"*)    recovery_steps+=("Gate 7: Load Skill(ring:dev-chaos-testing), dispatch ring:qa-analyst (test_mode=chaos, write only)") ;;
+      *"Gate 8"*)    recovery_steps+=("Gate 8: Load Skill(ring:codereview), dispatch ALL 8 reviewers in parallel") ;;
+    esac
+  done
+
+  # Deduplicate recovery steps
+  RECOVERY_JSON_ARRAY=$(printf '%s\n' "${recovery_steps[@]}" | sort -u | jq -R . | jq -sc .)
+
+  # Extract task context for recovery instructions
+  TASK_TITLE=$(echo "$NEW_STATE" | jq -r ".tasks[$CURRENT_TASK_INDEX].title // empty")
+  TASK_FILE=$(echo "$NEW_STATE" | jq -r ".tasks[$CURRENT_TASK_INDEX].subtasks[0].file // .source_file // empty")
```

Shellcheck (0.11.0)

[warning] TASK_FILE appears unused. Verify use (or export if used externally).

(SC2034)